cmd/process-agent/collector.go

```diff
@@ -315,6 +329,71 @@ func (l *Collector) run(exit chan struct{}) error {
 	return nil
 }
 
+func (l *Collector) resultsQueueForCheck(name string) *api.WeightedQueue {
+	switch name {
+	case checks.Pod.Name():
+		return l.podResults
+	case checks.Process.RealTimeName(), checks.RTContainer.Name():
+		return l.rtProcessResults
+	}
+	return l.processResults
+}
+
+func (l *Collector) runnerForCheck(c checks.Check, exit chan struct{}) (func(), error) {
+	results := l.resultsQueueForCheck(c.Name())
+
+	if withRealTime, ok := c.(checks.CheckWithRealTime); ok {
+		rtResults := l.resultsQueueForCheck(withRealTime.RealTimeName())
+
+		return checks.NewRunnerWithRealTime(
+			checks.RunnerConfig{
+				CheckInterval: l.cfg.CheckInterval(withRealTime.Name()),
+				RtInterval:    l.cfg.CheckInterval(withRealTime.RealTimeName()),
+
+				ExitChan:       exit,
+				RtIntervalChan: l.rtIntervalCh,
+				RtEnabled: func() bool {
+					return atomic.LoadInt32(&l.realTimeEnabled) == 1
+				},
+				RunCheck: func(options checks.RunOptions) {
+					l.runCheckWithRealTime(withRealTime, results, rtResults, options)
+				},
+			},
+		)
+	}
+	return l.basicRunner(c, results, exit), nil
+}
+
+func (l *Collector) basicRunner(c checks.Check, results *api.WeightedQueue, exit chan struct{}) func() {
```

Comment:
`legacyRunner`?

Reply:
It's not really legacy, maybe `simpleRunner`? because there are checks that will always be using this (e.g. there's no realtime`pod` or `connections`).

Reply:
Ah you are right, I missed the non-process checks.

Reply:
No problem, it would be nice to move some of this collector code under `pkg/process` including the runner code here. It's a bit busy.